bigquery/google/cloud/bigquery/client.py

```diff
@@ -667,6 +668,79 @@ def query_rows(self, query, job_config=None, job_id=None, timeout=None):
         job.begin()
         return job.result(timeout=timeout)
 
+    def list_rows(self, table, selected_fields=None,
+                  max_results=None, page_token=None, start_index=None):
+        """List the rows of the table.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/list
+
+        .. note::
+
+           This method assumes that the provided schema is up-to-date with the
+           schema as defined on the back-end: if the two schemas are not
+           identical, the values returned may be incomplete. To ensure that the
+           local copy of the schema is up-to-date, call ``client.get_table``.
+
+        :type table: One of:
+                     :class:`~google.cloud.bigquery.table.Table`
+                     :class:`~google.cloud.bigquery.table.TableReference`
+        :param table: the table to list, or a reference to it.
+
+        :type selected_fields: list of :class:`SchemaField`
+        :param selected_fields:
+            The fields to return. Required if ``table`` is a
+            :class:`~google.cloud.bigquery.table.TableReference`.
+
+        :type max_results: int
```

Comment:
I'm not seeing the max_results parameter getting passed to the iterator / extra params.

Reply:
Good catch, fixed.

Reply:
Thanks. I think there's a problem with this, though. Commented below. To get the correct behavior we need to either pass it in as `extra_params` or maybe remove some logic to cap results at one page from `HTTPIterator`.